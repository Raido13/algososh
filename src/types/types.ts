import { Dispatch, SetStateAction } from "react"

export type TFormState = {[key:string]: string};
export type TSetActive = Dispatch<SetStateAction<boolean>>;
export type TSetFormState = Dispatch<SetStateAction<{[key:string]: string}>>;