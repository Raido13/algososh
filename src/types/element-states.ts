export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified"
}

export enum LoaderStates {
  Add = 'add',
  Remove = 'remove',
  Reset = 'reset',
  Ascending = 'ascending',
  Descending = 'descending',
  AddHead = 'addHead',
  AddTail = 'addTail',
  RemoveHead = 'removeHead',
  RemoveTail = 'removeTail',
  AddIdx = 'addIdx',
  RemoveIdx = 'removeIdx'
}

export enum BorderStates {
  Default = '4px solid rgb(0, 50, 255)',
  Changing = '4px solid rgb(210, 82, 255)',
  Modified = '4px solid rgb(127, 224, 81)'
}