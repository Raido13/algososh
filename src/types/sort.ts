export enum Sort {
  Bubble = 'bubble',
  Choise = 'Choise'
}