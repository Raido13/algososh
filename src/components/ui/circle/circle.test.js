import { ElementStates } from "../../../types/element-states";
import { Circle } from "./circle";
import { render } from '@testing-library/react';

describe('Компонент Circle', () => {
  test('Отрисовка круга: Без текста', () => {
    const {asFragment} = render(<Circle />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка круга: С текстом', () => {
    const {asFragment} = render(<Circle letter="1" />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });
  
  test('Отрисовка круга: С Head', () => {
    const {asFragment} = render(<Circle head={'head'} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка круга: С Tail', () => {
    const {asFragment} = render(<Circle tail={'tail'} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка круга: С элементом в Head', () => {
    const {asFragment} = render(<Circle head={<Circle isSmall={true} letter='1' />}/>)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка круга: С элементом в Tail', () => {
    const {asFragment} = render(<Circle tail={<Circle isSmall={true} letter='1' />} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка круга: С индексом', () => {
    const {asFragment} = render(<Circle index={'1'} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка круга: В уменьшенной версии', () => {
    const {asFragment} = render(<Circle isSmall={true} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка круга: В дефолтном состоянии', () => {
    const {asFragment} = render(<Circle state={ElementStates.Default} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка круга: В изменяющемся состоянии', () => {
    const {asFragment} = render(<Circle state={ElementStates.Changing} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });

  test('Отрисовка круга: В модифицированном состоянии', () => {
    const {asFragment} = render(<Circle state={ElementStates.Modified} />)
    expect(asFragment().firstChild).toMatchSnapshot();    
  });
})