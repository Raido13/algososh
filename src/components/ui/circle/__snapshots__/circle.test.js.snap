// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Circle Отрисовка круга: Без текста 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга: В дефолтном состоянии 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга: В изменяющемся состоянии 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   changing"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга: В модифицированном состоянии 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   modified"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга: В уменьшенной версии 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle  small default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга: С Head 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга: С Tail 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Компонент Circle Отрисовка круга: С индексом 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга: С текстом 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    >
      1
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга: С элементом в Head 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle  small default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        >
          1
        </p>
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Отрисовка круга: С элементом в Tail 1`] = `
<div
  class="content "
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle  small default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        >
          1
        </p>
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;
